src/ray/raylet/node_manager.cc

```diff
@@ -736,6 +741,73 @@ void NodeManager::DispatchTasks(
   local_queues_.MoveTasks(assigned_task_ids, TaskState::READY, TaskState::RUNNING);
 }
 
+std::pair<std::shared_ptr<Worker>, bool> NodeManager::GetWorker(
+    const WorkerID &worker_id) {
+  bool is_worker = true;
+  auto worker = worker_pool_.GetRegisteredWorker(worker_id);
+  if (!worker) {
+    is_worker = false;
+    worker = worker_pool_.GetRegisteredDriver(worker_id);
+    if (!worker) {
+      return std::make_pair(nullptr, true);
+    }
+  }
+  return std::make_pair(worker, is_worker);
+}
+
+#define PREPROCESS_REQUEST(REQUEST_TYPE)                                            \
+  WorkerID worker_id = WorkerID::FromBinary(request.worker_id());                   \
```

Comment:
Better also add `request` as a parameter of this macro. Because otherwise, it's unclear what this `request` is.